python/cuml/cluster/kmeans.pyx

```diff
@@ -385,11 +379,11 @@ class KMeans(Base):
                             ' passed.')
 
         self.handle.sync()
+        print(self.cluster_centers_)
         cc_df = cudf.DataFrame()
-        for i in range(0, self.n_cols):
-            n_c = self.n_clusters
-            n_cols = self.n_cols
-            cc_df[str(i)] = self.cluster_centers_[i:n_c*n_cols:n_cols]
+        cc_df = cc_df.from_gpu_matrix(
+            self.cluster_centers_.reshape(self.n_clusters,
```

Comment:
I found this flow confusing that we assign to self.cluster_centers, then rearrange that data and reassign to self.cluster_centers. Would be better to keep it as a local until we're actually ready to do this final assignment in line 387.

Reply:
this is just simplifying the for loop above (which might have been one of the first 100 lines of cython code in cuML a long time ago!), unfortunately this is a straggler algorithm that still uses a cudf Series for storing the centers. Will be dealt soon in a PR that harmonizes all internal data structures, probably following similar work to cuDF's new use of RMM deviceBuffer.

Reply:
The reshaping is needed (it doesn't change the data, just changes the shape from a single dimensional array to a n_clusters x n_cols matrix that cuDF needs)

Reply:
I think the reshaping is not a big issue at all, just would prefer not assigning to a self field until it's actually in the right shape so then we maintain the invariant that self.cluster_centers_ either holds valid centers or nothing at all.

Reply:
Let's approve this now and push it in to unblock the queued up PR but try to refresh this a bit when converting it away from using cudf.DF/series to store centers.